bark/runtime/viewer/viewer.py

```diff
@@ -447,29 +450,80 @@ def drawLTLDebugInfomation(self, world, agent_id):
 
     def drawRssDebugInfomation(self, world, agent_id):
         from bark.core.world.evaluation import EvaluatorRSS
+        rss_responses = None
+        
         for evaluator in world.evaluators:
           if isinstance(world.evaluators[evaluator], EvaluatorRSS):
-            rss_responses = world.evaluators[evaluator].PairwiseDirectionalEvaluate(
-                world)
+            rss_responses = world.evaluators[evaluator].PairwiseDirectionalEvaluate(world)
             break
 
+        evaluator = None
+
+        if not rss_responses:
+          evaluator = EvaluatorRSS(agent_id = agent_id, params = self.parameters)
+          rss_responses = evaluator.PairwiseDirectionalEvaluate(world)
+
         def char_func(value):
-            if value == True:
+            if value is True:
                 return "T"
-            elif value == False:
+            elif value is False:
                 return "F"
+            elif abs(value) > 1000:
+                return "n.d."                
             else:
-                return "UNKNOWN"
+                return '%.2f' % value
 
         overall_safety = True
         if rss_responses:
-            self.drawText(position=(0.82, 0.91), text="ID  Lon  Lat")
+            self.drawText(position=(0.82, 0.91), text="ID  Lon  Lat   LonD   LatRightD LatLeftD")
             for i, (id, responses) in enumerate(rss_responses.items()):
-                overall_safety = overall_safety and any(responses)
-                str = "{}:    {}     {}".format(
+                (lon, lat, _, _, _) = responses
+                overall_safety = overall_safety and any((lon, lat))
+                str = "{}:    {}     {}     {}     {}      {}".format(
                     id, *list(map(char_func, responses)))
                 self.drawText(position=(0.82, 0.88-0.03*i), text=str)
 
+                # draw lat lon rectangle
+                lon_distance = responses[2]
+                lat_right_distance = responses[3]
+                lat_left_distance = responses[4]
+
+                ego_agent = world.agents[agent_id]
+                shape = ego_agent.shape
+
+                if self.invert_rss_polygon: # polygon appears right of car (Chinese Map)
+
+                  lat_lon_rectangle = GenerateGoalRectangle(
```

Comment:
Why is the goal rectangle being used to generate the Polygon? I thought we said we would base it on the lane-corridors.